app/src/main/java/org/simple/clinic/scheduleappointment/ScheduleAppointmentSheetController.kt

```diff
@@ -78,115 +65,14 @@ class ScheduleAppointmentSheetController @AssistedInject constructor(
-        .sorted()
-  }
-
-  private fun scheduleAppointments(
-      events: Observable<UiEvent>
-  ): Observable<UiChange> {
-
-    return Observable
-        .merge(
-            scheduleDefaultAppointmentDateForSheetCreates(events),
-            scheduleOnNextConfiguredAppointmentDate(events),
-            scheduleOnPreviousConfiguredAppointmentDate(events),
-            scheduleOnExactDate(events)
-        )
-        .distinctUntilChanged()
-        .doOnNext { latestAppointmentScheduled = it }
-        .map { appointmentDate ->
-          { ui: Ui ->
-            ui.updateScheduledAppointment(appointmentDate.scheduledFor, appointmentDate.timeToAppointment)
-            enableIncrements(ui)
-            enableDecrements(ui)
-          }
-        }
-  }
-
-  private fun enableIncrements(ui: Ui) {
-    val areLaterPotentialAppointmentsAvailable = latestAppointmentScheduled < allPotentialAppointmentDates.last()
-
-    ui.enableIncrementButton(areLaterPotentialAppointmentsAvailable)
-  }
-
-  private fun enableDecrements(ui: Ui) {
-    val areEarlierPotentialAppointmentsAvailable = latestAppointmentScheduled > allPotentialAppointmentDates.first()
-
-    ui.enableDecrementButton(areEarlierPotentialAppointmentsAvailable)
-  }
-
-  private fun scheduleOnExactDate(
-      events: Observable<UiEvent>
-  ): Observable<PotentialAppointmentDate> {
-    return events
-        .ofType<AppointmentCalendarDateSelected>()
-        .map { it.selectedDate }
-        .map(::generatePotentialAppointmentDate)
-  }
-
-  private fun scheduleOnPreviousConfiguredAppointmentDate(
-      events: Observable<UiEvent>
-  ): Observable<PotentialAppointmentDate> {
-    return events
-        .ofType<AppointmentDateDecremented>()
-        .map { previousConfiguredAppointmentDate(latestAppointmentScheduled) }
-  }
-
-  private fun scheduleOnNextConfiguredAppointmentDate(
-      events: Observable<UiEvent>
-  ): Observable<PotentialAppointmentDate> {
-    return events
-        .ofType<AppointmentDateIncremented>()
-        .map { nextConfiguredAppointmentDate(latestAppointmentScheduled) }
-  }
-
-  private fun scheduleDefaultAppointmentDateForSheetCreates(events: Observable<UiEvent>): Observable<PotentialAppointmentDate> {
-    val protocolUuid = Observable.fromCallable { currentFacility.get().protocolUuid!! }
-    val protocolStream = protocolUuid.switchMap(protocolRepository::protocol)
-
-    val configTimeToAppointment = Observable.fromCallable { config.defaultTimeToAppointment }
-    val protocolTimeToAppointment = protocolStream.map { Days(it.followUpDays) }
-
-    val timeToAppointments = Observable
-        .concatArrayEager(configTimeToAppointment, protocolTimeToAppointment)
-        .subscribeOn(schedulers.io())
-
-    return Observables.combineLatest(
-        events.ofType<ScreenCreated>(),
-        timeToAppointments
-    ) { _, timeToAppointment ->
-      timeToAppointment
-    }
-        .map(::generatePotentialAppointmentDate)
-  }
-
```

Comment:
The mess that this class is 🤦

Reply:
Yes, please don't do a `git blame` because some person who must not be named wrote most of it.

Good that it's going away. 😂